master/internal/api/handler.go

```diff
@@ -39,8 +39,11 @@ func Route(handler func(c echo.Context) (interface{}, error)) echo.HandlerFunc {
 }
 
 // WebSocketRoute upgrades incoming requests to websocket requests.
-func WebSocketRoute(handler func(socket *websocket.Conn, c echo.Context) error) echo.HandlerFunc {
+func WebSocketRoute(handler func(socket *websocket.Conn, c echo.Context) error, enableCORS bool) echo.HandlerFunc {
 	return func(c echo.Context) error {
+		if enableCORS {
+			upgrader.CheckOrigin = func(r *http.Request) bool { return true }
+		}
```

Comment:
Skip checking CORS when `enableCors` set to `true` in master config
Reference: https://github.com/determined-ai/determined/blob/9e8cd68ced6e955904ab3fbc423d0e5d61ef863c/master/internal/core.go#L1090